d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -168,6 +169,32 @@ void watchXdsResource(String resourceName, ResourceWatcher watcher)
     });
   }
 
+  @Override
+  public void watchAllXdsResources(WildcardResourceWatcher watcher)
+  {
+    _executorService.execute(() ->
+    {
+      _log.info("Subscribing to wildcard for resource type: {}", watcher.getType());
+      WildcardResourceSubscriber subscriber = getWildcardResourceSubscriber(watcher.getType());
+      if (subscriber == null)
+      {
+        subscriber = new WildcardResourceSubscriber(watcher.getType());
+        _wildcardSubscribers.put(watcher.getType(), subscriber);
+
+        if (_adsStream == null && !isInBackoff())
+        {
+          startRpcStreamLocal();
+        }
+        if (_adsStream != null)
+        {
+          _adsStream.sendDiscoveryRequest(watcher.getType(), Collections.singletonList("*"));
```

Comment:
Question for my knowledge: if the base `ResourceSubscriber` is already capable of doing glob collection, why wouldn't having a regular resource subscriber and then calling `watchXdsResource("*", ResourceWatcher myWatcher)` work to subscribe to everything?

(vs implementing a specific `WildcardResourceSubscriber`)

Reply:
As I said in the description, we need to provide the `watchAllXdsResources` method directly instead of mixing up with the former resource subsciber logic.

The `watchAllXdsResources` will not use the glob collection (which may be caused some potential risk like before). And is easy to be used to subscribe all the resources directlly.

Reply:
Yeah basically it needs separate code path where one subscriber can listen to multiple resources at once. The glob collections are a special case because a glob collection actually represents a map, and the `XdsClient` exposes updates to the collection as such. This way consumers of the `XdsClient` don't have to know whether glob collections are enabled.